api/design/auser.go

```diff
@@ -0,0 +1,57 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package design
+
+import (
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("user", func() {
+	Description("The user service exposes endpoint to get user specific specs")
+
+	Error("invalid-code", ErrorResult, "Invalid Authorization code")
+	Error("invalid-token", ErrorResult, "Invalid User token")
+	Error("invalid-scopes", ErrorResult, "Invalid User scope")
+	Error("internal-error", ErrorResult, "Internal Server Error")
+
+	Method("RefreshAccessToken", func() {
+		Description("Refresh the access token of User")
+		Security(JWTAuth, func() {
+			Scope("refresh:token")
+		})
+		Payload(func() {
+			Token("refreshToken", String, "Refresh Token of User", func() {
+				Example("refreshToken", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9."+
+					"eyJleHAiOjE1Nzc4ODM2MDAsImlhdCI6MTU3Nzg4MDAwMCwiaWQiOjExLCJpc3MiOiJUZWt0b24gSHViIiwic2NvcGVzIjpbInJlZnJlc2g6dG9rZW4iXSwidHlwZSI6InJlZnJlc2gtdG9rZW4ifQ."+
+					"4RdUk5ttHdDiymurlZ_f7Uy5Pas3Lq9w04BjKQKRiCE")
+			})
+			Required("refreshToken")
+		})
+		Result(func() {
+			Attribute("data", accessToken, "User Access JWT")
+			Required("data")
+		})
+
+		HTTP(func() {
+			POST("/user/refresh/accesstoken")
+			Header("refreshToken:Authorization")
+
+			Response(StatusOK)
+			Response("internal-error", StatusInternalServerError)
```

Comment:
Are we missing invalid authorization code error ??

Reply:
why do we need `invalid authorization code error` ? this is not login api

Reply:
Then why are we mentioning it above ??

Reply:
ohh.. willl remove it :P